aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -362,22 +383,22 @@ struct ReduceOp {
     }
 
     using out_ptr_vec_t = at::detail::Array<out_scalar_t*, output_vec_size>;
-    using offset_vec_t = at::detail::Array<index_t, output_vec_size>;
+    using offset_vec_t = at::detail::Array<at::detail::Array<index_t, num_outputs + 1>, output_vec_size>;
     offset_vec_t base_offsets;
     out_ptr_vec_t out;
 
     #pragma unroll
     for (int i = 0; i < output_vec_size; i++) {
-      base_offsets[i] = output_calc.get(output_idx + i)[0];
-      out[i] = (out_scalar_t*)((char*)dst[0] + base_offsets[i]);
+      base_offsets[i] = output_calc.get(output_idx + i);
+      out[i] = (out_scalar_t*)((char*)dst[0] + base_offsets[i][0]);
```

Comment:
this is suspicious - how is this going to work for multiple outputs? Or multiple outputs case never takes this path?

Reply:
IIUC, this is used for storing intermediate accumulation data. And for accumulation, it stores arg_t, not output_scalar_t, and when accumulating using output tensor, it only uses the first output tensor.